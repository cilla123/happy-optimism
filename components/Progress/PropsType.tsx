export default interface PropsType {
  theme?: 'primary' | 'success' | 'warning' | 'error';
  strokeWidth?: number;
  percent?: number;
  shape?: 'line' | 'circle';
}
