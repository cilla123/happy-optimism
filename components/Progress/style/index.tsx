import '../../style/';
import '../../Spinner/style/';
import './index.scss';
