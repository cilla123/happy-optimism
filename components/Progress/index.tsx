import React, { PureComponent } from 'react';
import classnames from 'classnames';
import PropsType from './PropsType';
import Spinner from '../Spinner';

export interface ProgressProps extends PropsType {
  prefixCls?: string;
  className?: string;
}

export default class Progress extends PureComponent<ProgressProps, {}> {

  static defaultProps = {
    prefixCls: 'za-progress',
    theme: 'primary',
    shape: 'line',
    strokeWidth: 5,
    percent: 15,
  };

  render() {
    const { prefixCls, className, theme, shape, strokeWidth, percent, children, ...others } = this.props;

    const cls = classnames(`${prefixCls}`, className, {
      [`theme-${theme}`]: !!theme,
      [`shape-${shape}`]: !!shape,
    });

    const innerRender = shape === 'circle'
      ? <div className={`${prefixCls}-inner`}>
          <Spinner theme={theme} strokeWidth={strokeWidth} percent={percent} />
        </div>
      : <div className={`${prefixCls}-inner`} style={{ height: strokeWidth }}>
          <div className={`${prefixCls}-bg`} style={{ width: `${percent}%` }} />
        </div>;

    return (
      <div className={cls} {...others}>
        {innerRender}
        {children && <div className={`${prefixCls}-text`}>{children}</div>}
      </div>
    );
  }
}
