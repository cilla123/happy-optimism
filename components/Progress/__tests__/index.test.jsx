import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Progress from '../index';

describe('Progress', () => {
  it('renders correctly', () => {
    const wrapper = render(<Progress percent={10} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders has children correctly', () => {
    const wrapper = render(<Progress percent={10}>foo</Progress>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders shape is circle correctly', () => {
    const wrapper = render(<Progress shape="circle" percent={10}>foo</Progress>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
