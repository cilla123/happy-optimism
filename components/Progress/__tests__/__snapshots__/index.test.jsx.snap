// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress renders correctly 1`] = `
<div
  class="za-progress theme-primary shape-line"
>
  <div
    class="za-progress-inner"
    style="height:5px"
  >
    <div
      class="za-progress-bg"
      style="width:10%"
    />
  </div>
</div>
`;

exports[`Progress renders has children correctly 1`] = `
<div
  class="za-progress theme-primary shape-line"
>
  <div
    class="za-progress-inner"
    style="height:5px"
  >
    <div
      class="za-progress-bg"
      style="width:10%"
    />
  </div>
  <div
    class="za-progress-text"
  >
    foo
  </div>
</div>
`;

exports[`Progress renders shape is circle correctly 1`] = `
<div
  class="za-progress theme-primary shape-circle"
>
  <div
    class="za-progress-inner"
  >
    <svg
      class="za-spinner theme-primary"
      viewBox="0 0 62 62"
    >
      <circle
        class="za-spinner-path"
        cx="31"
        cy="31"
        fill="none"
        r="28.5"
        style="stroke-width:5"
      />
      <circle
        class="za-spinner-line"
        cx="31"
        cy="31"
        fill="none"
        r="28.5"
        style="stroke-dasharray:17.90707812546182 179.0707812546182;stroke-width:5"
      />
    </svg>
  </div>
  <div
    class="za-progress-text"
  >
    foo
  </div>
</div>
`;
